sqlglot/helper.py

```diff
@@ -7,40 +7,57 @@
 from copy import copy
 from enum import Enum
 
+from sqlglot.expressions import Expression, Table
+
 CAMEL_CASE_PATTERN = re.compile("(?<!^)(?=[A-Z])")
 logger = logging.getLogger("sqlglot")
 
 
 class AutoName(Enum):
-    def _generate_next_value_(name, _start, _count, _last_values):
+    def _generate_next_value_(name: str, _start: int, _count: int, _last_values: int) -> str:
         return name
 
 
-def list_get(arr, index):
+def list_get(arr: t.List[t.Any], index: int) -> t.Optional[t.Any]:
+    """Returns the value in `arr` at position `index`, or `None` if `index` is out of bounds."""
     try:
         return arr[index]
     except IndexError:
         return None
 
 
-def ensure_list(value):
+def ensure_list(value: t.Any) -> t.Union[t.List[t.Any], t.Tuple[t.Any], t.Set[t.Any]]:
```

Comment:
The fact that this doesn't actually ensure a list type becomes even more confusing when we start adding types. Since we don't like the cost of casting to list can we rename to `ensure_iterable`? Also should we establish a term in the codebase for referencing an iterable that is not a str or byte and then use that in the name? Maybe `ensure_collection`?

Reply:
ensure_collection is fine

Reply:
Agreed -- I had some trouble deciding on a name for this function. I wasn't sure what was the intent we wanted to capture because there are generally two uses throughout the codebase:

1. Convert something into a list, because we expect it to be a list in other parts of the code.
2. Convert something into a list only so that we can iterate over it.

If we do `ensure_iterable`, it might obscure the intent for case 1.

Reply:
This might be a pain but my vote would be to have `ensure_list` and `ensure_collection` and then functions would use the one appropriate for their use case. For example in the DataFrame implementation I really do want a list so I would use `ensure_list` (and take the performance consequences) while other places we might actually be fine with a collection.

Reply:
Got it, I'll try that.